pkg/cmd/roachtest/spec/machine_type.go

```diff
@@ -49,26 +52,37 @@ func AWSMachineType(cpus int, highmem bool) string {
 	return fmt.Sprintf("%s.%s", family, size)
 }
 
-// GCEMachineType selects a machine type given the desired number of CPUs.
-func GCEMachineType(cpus int, highmem bool) string {
+// GCEMachineType selects a machine type given the desired number of CPUs and
+// memory per CPU ratio.
+func GCEMachineType(cpus int, mem MemPerCPU) string {
 	// TODO(peter): This is awkward: at or below 16 cpus, use n1-standard so that
 	// the machines have a decent amount of RAM. We could use custom machine
 	// configurations, but the rules for the amount of RAM per CPU need to be
 	// determined (you can't request any arbitrary amount of RAM).
 	series := "n1"
-	kind := "standard" // 3.75 GB RAM per CPU
-	if highmem {
+	var kind string
+	switch mem {
+	case Auto:
```

Comment:
what's the motivation behind the auto option? I.e. why with more CPU's should the ram to cpu ratio be higher, if mem is not specified? If we do decide to keep the auto option, I think we'll need to add casing to aws as well.

Reply:
auto is the current state :)
with more cores you get less ram per core, I asked Renato about this here https://cockroachlabs.slack.com/archives/C023S0V4YEB/p1677709976637079

Reply:
It seems the abstraction we're trying to build here with these memory requirements is inherently leaky, but as @msbutler says, we should try to have the the same behavior for AWS for `Auto`.